### ISSUE TITLE
Run servers on demand

### ISSUE THREAD

**Issue by Aidoboy** [GameServerManagers/LinuxGSM#1465]
Would it be possible to suspend empty servers, and restart them when players try to connect? It would mean much more could be run on one machine, but it might not be possible. I'm specifically thinking about Minecraft and Source games.

Edit: This could be relevant: https://www.planetminecraft.com/blog/automatically-stop-a-server-when-nobodys-playing/

**Comment by MarkL4YG** [comment]
I could think of more sophisticated ways of doing this for Minecraft by for example getting a real playercount from the server by pinging it rather than just sending a command and hoping that the last 6 lines in the logfile(!) will be your answer.
I also think bothering with shutting down Minecraft servers doesn't earn you anything - it might in fact even worsen performance as you completely undo the optimizations of the JVM by shutting it down instead of letting it idle. If you want to reduce load when nobody is connected: Don't use chunkloaders and switch off automatic loading of the spawn chunks. You should see CPU time dropping to a minimum as only scheduled Tasks are run and no chunk is actually loaded. If you're talking about shutting down minigame servers: You may be off better by using an actual BungeeCord plugin for this. 

Regarding other games: If there's a common way to determine the amount of players online - sure why not. If not - then I'll expect this to be rejected since there doesn't seem to be an urgent reason for this. But I guess you may get better support on this in the forums of each specific game.

**Comment by UltimateByte** [comment]
CS servers already go to half sleep when they're idle to save CPU. CPU is the most valuable thing on a game server machine.

Let's say your idea is a good idea, which i don't think, to allow such a connection to an offline server, we should run a process that simulates server listening, which is almost impossible to make properly, and answers to queries, then the client should be able to wait until the server is up to initiate a real connection instead of time out.

Basically there is no way it works and the point of a server is to never be down anyways.
